src/components/Modal/Modal.js

```diff
@@ -73,6 +77,12 @@ class Modal extends PureComponent<Props, State> {
     }
   }
 
+  handleKeyboardEvent = (event: any) => {
+    if (event.keyCode === 27) {
```

Comment:
Nit. Escape handling is working but I would use `if (event.key === 'Escape') { }`.
We could also leave it with the `keyCode` as it's only one key here but with the string, it's more readable.